application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -701,6 +710,28 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                         } catch (Exception e) {
                         }
 
+                        PageLink pageLink = new PageLink(1000);
+                        PageData<User> users;
+                        do {
+                            List<ListenableFuture<?>> futures = new ArrayList<>();
+                            users = userService.findAllUsers(pageLink);
+                            for (User user : users.getData()) {
+                                futures.add(dbUpgradeExecutor.submit(() -> {
+                                    JsonNode additionalInfo = user.getAdditionalInfo();
```

Comment:
Upgrade failure for one user should not affect other users or upgrade procedure. Make sure you add `try-catch` block for individual user upgrade. Log the user id in case of failure but don't stop the update.